tensorflow/core/kernels/data/map_dataset_op_test.cc

```diff
@@ -0,0 +1,363 @@
+  // kernel and context internally.
+  DatasetBase* CreateMapDataset(DatasetBase* input_dataset,
+                                const string& func_name) {
+    map_inputs_.clear();
+    map_kernel_ =
+        CreateMapDatasetOpKernel<int64>(input_dataset->name(), func_name);
+
+    // Save the input dataset into a variant tensor as the input of MapDataset.
+    Tensor dataset_tensor(DT_VARIANT, TensorShape({}));
+    TF_CHECK_OK(StoreDatasetInVariantTensor(input_dataset, &dataset_tensor));
+    Variant variant = dataset_tensor.scalar<Variant>()();
+    TF_CHECK_OK(AddDatasetInputFromArray<Variant>(
+        &map_inputs_, map_kernel_->input_types(), TensorShape({}), {variant}));
+
+    TF_CHECK_OK(
+        CreateOpKernelContext(map_kernel_.get(), &map_inputs_, &map_context_));
+    TF_CHECK_OK(CheckOpKernelInput(*map_kernel_, map_inputs_));
+    DatasetBase* map_dataset;
+    TF_CHECK_OK(
+        CreateDataset(map_kernel_.get(), map_context_.get(), &map_dataset));
+    return map_dataset;
+  }
+
+ private:
+  // Creates a new MapDataset op kernel. The `input_dataset` parameter should be
+  // same with the node name of the input dataset for the method
+  // `CreateMapDataset()`.
+  template <typename T>
+  std::unique_ptr<OpKernel> CreateMapDatasetOpKernel(
+      const string& input_dataset, const string& func_name) {
+    DataTypeVector map_output_dtypes({tensorflow::DataTypeToEnum<T>::value});
+    gtl::ArraySlice<TensorShape> map_output_shapes({{}});
+    FunctionDefHelper::AttrValueWrapper func =
+        FunctionDefHelper::FunctionRef(func_name, {{"T", DT_INT64}});
+
+    map_node_def_ = test::function::NDef(kNodeName, kOpName, {input_dataset},
+                                         {{"f", func},
+                                          {"Targuments", {}},
+                                          {"output_shapes", map_output_shapes},
+                                          {"output_types", map_output_dtypes},
+                                          {"use_inter_op_parallelism", true},
+                                          {"preserve_cardinality", false}});
+    std::unique_ptr<OpKernel> map_op_kernel;
+    TF_CHECK_OK(CreateOpKernel(map_node_def_, &map_op_kernel));
+    return map_op_kernel;
+  }
+
+ protected:
+  DatasetBase* range_dataset_;
+  DatasetBase* map_dataset_;
+  std::unique_ptr<OpKernel> map_kernel_;
+  std::unique_ptr<OpKernelContext> map_context_;
+  std::unique_ptr<IteratorContext> iterator_context_;
+  std::unique_ptr<IteratorBase> iterator_;
+  std::unique_ptr<SerializationContext> serialization_context_;
+
+ private:
+  NodeDef map_node_def_;
+  gtl::InlinedVector<TensorValue, 4> map_inputs_;
+};
+
+const char* MapDatasetOpTest::kNodeName = "map_dataset";
+const char* MapDatasetOpTest::kOpName = "MapDataset";
+
+struct DatasetGetNextTest
+    : MapDatasetOpTest,
+      ::testing::WithParamInterface<
+          std::tuple<int64, int64, int64, string, std::vector<int64>,
```

Comment:
Instead of using a tuple, create a struct to represent the test case. It will make the code easier to read.